[Martin Ng] iP

martinng01 opened the pull request:
# Jun Jie the Chatbot
> “from today onwards, you are jun jie.” – David Allen ([source](https://www.youtube.com/watch?v=GHiyPYaMYrM))

Jun Jie can help you remember your tasks for the day, but don't mess around because he's 

- *OFFENSIVE*
- RUDE :cursing_face:
- Zero tolerance for incorrect inputs :skull:

All you need to do is,

1. download it from [here](https://github.com/martinng01/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let jun jie manage your tasks for you 😉

And it is *FREE!*

Features:

- [x] Managing tasks
- [x] Managing deadline
- [ ] GUI (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

martinng01 committed:
rename bot

martinng01 committed:
implement echo

martinng01 committed:
add commands

martinng01 committed:
implemented marking doneness

martinng01 committed:
add task types

martinng01 committed:
add automated texting

martinng01 committed:
added error handling, delete

martinng01 committed:
add javadocs

martinng01 committed:
Add saving and loading of data

martinng01 committed:
Add date-time parsing

martinng01 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/DeadlineTask.java
#	src/main/java/EventTask.java

martinng01 committed:
Make the code more OOP

martinng01 committed:
Divide classes into packages

martinng01 committed:
Merge branch 'add-gradle-support' of https://github.com/martinng01/ip

martinng01 committed:
Automate project builds using Gradle

martinng01 committed:
Add JUnit Tests

martinng01 committed:
Improve OOP

martinng01 committed:
Add JavaDoc comments

martinng01 committed:
Comply code with coding standard

martinng01 committed:
Fix parser

martinng01 committed:
Add find command

martinng01 committed:
Merge branch 'branch-Level-9'

martinng01 committed:
Merge branch 'branch-A-CodingStandard'

martinng01 committed:
Resolve conflicts

martinng01 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/tasks/DeadlineTask.java

ljy0422 reviewed (commented):
After a thorough review of the code, no obvious violations of standard Java coding style were found. Overall, your code exhibits strong compliance with the coding standard. Keep up with the good work! > <

martinng01 committed:
Rename duke package to junjie package

martinng01 committed:
Add a GUI to the chatbot

martinng01 committed:
Merge branch 'branch-Level-10'

tanjiajiajun commented on src/main/java/junjie/Ui.java:
```diff
@@ -0,0 +1,20 @@
+     * @param str The string to be printed.
+     */
+    public void print(String str) {
+        System.out.println("-------------------------");
+        System.out.println(str);
+        System.out.println("-------------------------");
```
Can consider adding a reusable final String LINE = "-------------------------"

tanjiajiajun commented on src/main/java/junjie/tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    @Override
+    public String toFileString() {
+        return String.format("D | %d | %s | %s", isDone() ? 1 : 0, getName(), deadline);
```
A very clean way to implement the string formatting!

tanjiajiajun commented on src/main/java/junjie/tasks/Task.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public abstract class Task {
+    private final String name;
+    private boolean done = false;
```
can be changed to isDone because it is a boolean value

tanjiajiajun reviewed (commented):
Looks good to merge. Just some nits. Overall high coding standards, good job!

martinng01 committed:
Use assertions

Assertions were included to check things that should never happen.
For example, parameters for tasks should not be null.

martinng01 committed:
Update code quality

Extracted createTaskFromDetails function from TaskList constructor,
 so that the constructor is not too long.

martinng01 committed:
Merge pull request #2 from martinng01/branch-A-CodeQuality

Update code quality

martinng01 committed:
Merge branch 'master' into branch-A-Assertions

martinng01 committed:
Merge pull request #3 from martinng01/branch-A-Assertions

Branch a assertions

martinng01 committed:
Add ability to tag tasks.

This way tasks can be classified together and user can find tasks
more easily.

martinng01 committed:
Merge branch 'branch-C-Tagging'

martinng01 committed:
Add user guide.

martinng01 committed:
Bugfix for tasks with no tags crashing the application

martinng01 committed:
Update tests to include testing of tags
